examples/ddp_00.py

```diff
@@ -0,0 +1,50 @@
+from dataclasses import dataclass, field
+import logging
+import os
+from typing import Any, List
+
+import hydra
+from hydra.core.config_store import ConfigStore
+from omegaconf import DictConfig, MISSING
+import torch
+import torch.distributed as dist
+
+log = logging.getLogger(__name__)
+
+
+defaults = [
+    {"hydra/launcher": "joblib"}
+]
+
+@dataclass
+class DDPConf:
+    defaults: List[Any] = field(default_factory=lambda: defaults)
+    backend: str = "gloo"
+    init_method: str = MISSING
+    world_size: int = 4
```

Comment:
```suggestion
    nodes: int = 1
    gpus_per_node: int = 2
    world_size: int = nodes * gpus_per_node
```

Similarly, what about doing this?

Reply:
my only concern will be, gpus_per_node is an implementation detail :) I need 8 gpus. Wether they come from 2 nodes (4 gpus each) or 3 nodes (2, 2, 4 gpus) doesnt make any functional difference to DDP.

Reply:
good point.

For some reason in the past I thought it was helpful to have more control over the distribution of gpus per node, but I can't recall why and it seems it makes no difference to `init_method`.

So you can probably disregard this particular suggestion!